src/instructlab/model/evaluate.py

```diff
@@ -33,889 +32,768 @@ class Benchmark(str, enum.Enum):
-                    tls_client_passwd=ctx.params["tls_client_passwd"],
-                    tls_insecure=ctx.params["tls_insecure"],
-                    model=model,
-                    model_name=model_name,
-                    max_workers=max_workers,
-                    gpus=gpus,
-                    backend=backend,
-                    enable_serving_output=enable_serving_output,
-                )
-                evaluator.gen_answers(
-                    api_base, max_workers=max_workers, serving_gpus=effective_gpus
-                )
-            finally:
-                if server is not None:
-                    server.shutdown()
+            raise click.exceptions.Exit(1)
 
-            print("Evaluating answers...")
-            try:
-                server, api_base, effective_gpus = launch_server(
-                    eval_serve=ctx.obj.config.serve,
-                    tls_client_cert=ctx.params["tls_client_cert"],
-                    tls_client_key=ctx.params["tls_client_key"],
-                    tls_client_passwd=ctx.params["tls_client_passwd"],
-                    tls_insecure=ctx.params["tls_insecure"],
-                    model=judge_model,
-                    model_name=judge_model_name,
-                    max_workers=max_workers,
-                    gpus=gpus,
-                    backend=judge_backend,
-                    enable_serving_output=enable_serving_output,
-                )
-                overall_score, qa_pairs, turn_scores, error_rate = (
-                    evaluator.judge_answers(
-                        api_base, max_workers=max_workers, serving_gpus=effective_gpus
-                    )
-                )
-            finally:
-                if server is not None:
-                    server.shutdown()
+        validate_model(model)
+        validate_model(judge_model, "--judge-model")
+        if benchmark == Benchmark.MT_BENCH_BRANCH:
+            validate_model(base_model, "--base-model")
 
-            max_score = get_benchmark_max_score(Benchmark.MT_BENCH)
-            print("# SKILL EVALUATION REPORT")
-            print("\n## MODEL (SCORE)")
-            display_model(model, overall_score, max_score)
-            print(f"\n### TURN ONE (0.0 to {max_score}):")
-            print(round(turn_scores[0], 2))
-            print(f"\n### TURN TWO (0.0 to {max_score}):")
-            turn2_score = turn_scores[1]
-            if isinstance(turn2_score, float):
-                turn2_score = round(turn2_score, 2)
-            print(turn2_score)
-            display_error_rate(error_rate)
+        if (isinstance(max_workers, str) and max_workers != "auto") or (
+            isinstance(max_workers, int) and max_workers < 1
+        ):
+            click.secho(
+                "max-workers must be specified as a positive integer or 'auto'",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
```

Comment:
if this is raising an exception, do we need do catch this?

Reply:
wait also, we can't raise a click exception inside of this pkg, we should have no references to click.

this should raise a generic exception and `src/instructlab/cli/model/evaluate.py` should have a try except that raises a click.exceptions.Exit(1)